solana/pyth2wormhole/client/src/main.rs

```diff
@@ -660,12 +673,13 @@ async fn attestation_sched_job(args: AttestationSchedJobArgs) -> Result<(), ErrB
         let group_name4err_msg = batch.group_name.clone();
 
         // We never get to error reporting in daemon mode, attach a map_err
-        let job_with_err_msg = job.map_err(move |e| {
+        let job_with_err_msg = job.or_else(move |e| async move {
             warn!(
                 "Batch {}/{}, group {:?} ERR: {:#?}",
                 batch_no4err_msg, batch_count4err_msg, group_name4err_msg, e
             );
-            e
+            HEALTHCHECK_STATE.lock().await.add_result(false); // Report this job as failed to healthcheck
+            Err(e)
```

Comment:
I think you should extract out a function like `attestation_job_with_healthcheck` that wraps `attestation_job` and adds both cases of the healthcheck logic. That approach reduces the probability of programmer errors compared to splitting the logic out into two different places in the code.

Reply:
I think an async block inside `attestation_job` could work well for this. As for the separation, it's the non-daemon mode that's making it necessary. I don't think we need the daemon mode anymore. It was only ever used for triggering readiness in the Python script and this changeset makes the Python readiness socket obsolete.

Reply:
That said, it deserves its own change better IMO

Reply:
I agree that removing the non-daemon mode is the way to go (and makes code much simpler), specially after this change. I thin it's better to do it in a subsequent PR and not here.